### ISSUE TITLE
Unable to parse core.php

### ISSUE THREAD

**Issue by skipthedrive** [wpsharks/comet-cache#549]
Hello -

I've been receiving a lot of warnings/errors in my PHP log file, and I believe it's due to manually clearing the cache. Please see below:

```
PHP Fatal error:  Uncaught exception 'Exception' with message 'Unable to parse: `http://97.74.25.xxx connection: keep-alive/`.' in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php:3236
Stack trace:
#0 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php(3450): WebSharks\HtmlCompressor\Core->mustParseUrl('http://97.74.25...')
#1 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php(1339): WebSharks\HtmlCompressor\Core->resolveRelativeUrl('http://www.skip...')
#2 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php(1154): WebSharks\HtmlCompressor\Core->compileJsTagFragsIntoParts(Array, 'head')
#3 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php on line 3236

[13-Aug-2015 14:41:54 UTC] PHP Warning:  mkdir(): File exists in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php on line 2565

[13-Aug-2015 14:45:39 UTC] PHP Warning:  mkdir(): File exists in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php on line 2565

[13-Aug-2015 14:45:39 UTC] PHP Fatal error:  Uncaught exception 'Exception' with message 'Cache directory not readable/writable: `/home/content/##/########/html/wp-content/cache/zencache/htmlc/private`. Failed on `/home/content/##/########/html/wp-content/cache/zencache/htmlc/private/www-skipthedrive-com/e/1/b/e/0`.' in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php:2578
Stack trace:
#0 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php(1308): WebSharks\HtmlCompressor\Core->cacheDir('private', 'e1be0bae546a706...')
#1 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php(1154): WebSharks\HtmlCompressor\Core->compileJsTagFragsIntoParts(Array, 'head')
#2 /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php(347): WebSharks\HtmlCompressor\Core->may in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-compressor/src/includes/classes/Core.php on line 2578

[13-Aug-2015 13:14:56 UTC] PHP Warning:  mkdir(): File exists in /home/content/##/########/html/wp-content/plugins/zencache-pro/src/vendor/websharks/html-
```

**Comment by raamdev** [comment]
@skipthedrive Thank you for the report. Can you tell me which version of PHP you're running, along with which version of ZenCache Pro, and which web server (e.g., Apache)?

@jaswsinc Does this look like an issue with the HTML Compressor parsing requests containing IP addresses?

**Comment by skipthedrive** [comment]
Hi @raamdev -
I'm using PHP version 5.4.19, ZenCache Pro version 150716, and Apache.
Thanks,
Pete

**Comment by jaswrks** [comment]
> Does this look like an issue with the HTML Compressor parsing requests containing IP addresses?

No, I don't think so. The HTML Compressor uses PHP's own `parse_url()` method, and this correctly parses IP host names in all versions of PHP: https://3v4l.org/RVEdj

---

Inside the backticks is interesting:

> Unable to parse: `http://97.74.25.xxx connection: keep-alive/`

The HTML Compressor was literally unable to parse:

```
http://97.74.25.xxx connection: keep-alive/
```

and that's to be expected. What's not expected is that we would attempt to parse a URL like this to begin with. It suggests there is a problem in parsing the HTML markup itself, because that's not a valid URL.

@skipthedrive Can you provide a URL which links us to the page that is being compressed please?

**Comment by skipthedrive** [comment]
Hey @jaswsinc -

I believe my homepage is being compressed, since it's not excluded from being cached. Is this a correct assumption? My homepage is SkipTheDrive (dot) com.

I have Yes selected for all HTML compression, except for combining CSS from head and body into fewer files. I remember writing in about this a while back. For some reason when this is enabled, it messes up my site layout.

**Comment by jaswrks** [comment]
> except for combining CSS from head and body into fewer files

Copy that. We have this issue resolved in the development copy, and it should get pushed out in a new release of ZenCache Pro soon. So at some point in the near future you should be able to enable this again. Referencing: https://github.com/websharks/html-compressor/releases/tag/150821

> I believe my homepage is being compressed, since it's not excluded from being cached. Is this a correct assumption? My homepage is SkipTheDrive (dot) com.

Yes, it is being compressed. Is that the page that is triggering the PHP exception?

![2015-08-21_00-21-08](https://cloud.githubusercontent.com/assets/1563559/9404275/9ac736bc-479a-11e5-9fce-fe741989e45c.png)

What I'm trying to determine is where this URL comes from:

```
Unable to parse: `http://97.74.25.xxx connection: keep-alive/`
```

Note that the HTML Compressor is trying to connect to: `http://97.74.25.xxx connection: keep-alive/`, which is totally bogus, because `connection: keep-alive/` is not part of a valid URL.

**Comment by skipthedrive** [comment]
@jaswsinc - I'm a bit confused. Aren't all of my pages (except the ones being excluded) compressed? Would the below steps help isolate the problematic page?
1.) Start with a clear cache
2.) Cache one page (at a time)
3.) Manually clear the cache and see if that was the problematic page

Please let me know.
Thanks,
Pete

**Comment by jaswrks** [comment]
> Aren't all of my pages (except the ones being excluded) compressed?

It would appear so, yes.

> Would the below steps help isolate the problematic page?
> 1.) Start with a clear cache
> 2.) Cache one page (at a time)
> 3.) Manually clear the cache and see if that was the problematic page

Almost, yes. Try it like this:
- Start with a clear cache.
- Hit one page on your site at a time, and after each page is cached (i.e., each time you load a page that gets cached) check your PHP error log or look for the error on-site.
- Continue that process (no need to clear the cache), until the error occurs.
- Record the URL on which the error occured and report that to us here.

**Comment by skipthedrive** [comment]
@jaswsinc - Thanks for the details. I might not get around to this for a few days, but will let you know once I do.

**Comment by skipthedrive** [comment]
Hey @jaswsinc - 

I noticed that over the past few days I haven't received these messages. Still, I decided to go ahead and walk through the steps you laid out above. I'm still not receiving any messages, so I suppose we could close this out if you'd like. I'm not sure what may have been causing the problem. I did recently upgrade WP to the latest version, but don't know if that may have helped or not.
